core/src/main/scala/spinal/core/AFix.scala

```diff
@@ -720,23 +719,23 @@ class AFix(val maxRaw: BigInt, val minRaw: BigInt, val exp: Int) extends MultiDa
    * Rounds a value up towards positive infinity at the given exp point position
    * @return Rounded result
    */
-  def ceil(exp : Int): AFix = {
+  def ceil(exp : Int): AFix = ceil(exp, getTrunc.align)
+
+  def ceil(exp: Int, aligned: Boolean): AFix = {
     val drop = exp-this.exp
     if(drop < 0) return CombInit(this)
     val step = BigInt(1) << drop
-    val res = new AFix((this.maxRaw+step-1) >> drop, (this.minRaw+step-1) >> drop, exp)
+    val res = new AFix(((this.maxRaw+step-1) >> drop)-1, (this.minRaw+step-1) >> drop, exp)
```

Comment:
Are you sure of the : 
((this.maxRaw+step-1) >> drop)-1
instead of 
(this.maxRaw+step-1) >> drop

?

I mean, let's take the example of a Q0.16 that we want to ceil to remove 4 lsb bits.
if we ceil the max value 0.FFFF, it would go to 1.000 => Q1.12 => 4096 maxRaw

(this.maxRaw+step-1) >> drop would give (0xFFFF+0x10-1) >> 4 = 4096  => OK
while 
((this.maxRaw+step-1) >> drop)-1 would give 4095 => KO

If i understand well the "aligned" i think it should be taken in acount there too. isn't it ?

Reply:
I got it. This part is rolled back.